### ISSUE TITLE
Parallelization of integral routines for PE/EFP

### ISSUE THREAD

**Issue by maxscheurer** [psi4/psi4#2007]
Currently the integrals required in polarizable embedding models (PE and EFP) don't exploit any parallelization.
This should be fixed (by me) once #1721 is in, because for larger systems, the computations become really slow.

Routines:
- `electric_field_value`
- `induction_operator`
- `ao_multipole_potential`

The last routine should also be tweaked to be called only once with all site positions.

**Comment by andysim** [comment]
Yes, the 1 electron integral routines are getting a very significant rewrite after that pull request is in; they will use libint2, and will be screened for overlap.  The original 1e integral code was really written with the standard core hamiltonian in mind, where performance is never an issue.  Parallelization of that rewritten code should be relatively easy.  I haven't quite decided the best order to reconcile the changes to the integral engine with #1946 yet, but we'll get that merged as soon as we can.  Thanks for posting this issue so that we can all coordinate - hopefully everything will be much easier when all of the old integral code is ripped out.

**Comment by maxscheurer** [comment]
Sounds good! It's not urgent at all... just wanted to leave this as a note that I don't forget. I'm happy to make a PR once the prerequesites are in! 😄
